services/QuillLMS/client/app/bundles/Diagnostic/components/questions/response.tsx

```diff
@@ -0,0 +1,439 @@
+import * as React from 'react'
+import * as jsDiff from 'diff'
+
+import { ContentState, EditorState } from 'draft-js'
+import _ from 'underscore'
+
+import {
+  Modal,
+  TextEditor
+} from '../../../Shared/index'
+import massEdit from '../../actions/massEdit'
+import questionActions from '../../actions/questions'
+import {
+  deleteResponse,
+  getGradedResponsesWithCallback,
+  submitResponseEdit,
+} from '../../actions/responses'
+import sentenceFragmentActions from '../../actions/sentenceFragments'
+import C from '../../constants'
+import { rematchOne } from '../../libs/grading/rematching'
+import { getStatusForResponse } from '../../../Shared/index'
+
+import ConceptSelectorWithCheckbox from '../shared/conceptSelectorWithCheckbox.jsx'
+
+import ResponseList from './responseList'
+import { Question, Response } from '../../libs/grading/rematching'
+
+interface ResponseProps {
+  allExpanded: boolean,
+  ascending: boolean,
+  concepts: Array<Object>,
+  dispatch: Function,
+  expand: Function,
+  expanded: boolean,
+  getChildResponses: Function,
+  getResponse: Function,
+  mode: string,
+  massEditResponses: Object,
+  passedResponse: Response,
+  question: Question,
+  questionID: string,
+  readOnly: boolean,
+  responses: Array<Response>,
+  state: Object,
+  states: Object,
+  updateParentResponse: Function,
+}
+
+
+const Response = ({allExpanded, ascending, concepts, dispatch, expand, expanded, getChildResponses, getResponse, massEditResponses, mode, passedResponse, question, questionID, readOnly, responses, state, states, updateParentResponse}: ResponseProps) => {
+  const [response, setResponse] = React.useState(passedResponse)
+  const [feedback, setFeedback] = React.useState(passedResponse.feedback || '')
+  const [selectedBoilerplate, setSelectedBoilerplate] = React.useState('')
+  const [parent, setParent] = React.useState(null)
+  const [conceptResults, setConceptResults] = React.useState(extractConceptResultsFromResponse(response))
+  const [actions, setActions] = React.useState(mode === 'sentenceFragment' ? sentenceFragmentActions : questionActions)
+  const [statusCode, setStatusCode] = React.useState(getStatusForResponse(passedResponse))
+  const newResponseOptimal = React.useRef(null)
+
+  React.useEffect(() => {
+    setFeedback(response.feedback)
+    setStatusCode(getStatusForResponse(response))
+    setConceptResults(extractConceptResultsFromResponse(response))
+    setStatusCode(getStatusForResponse(response))
+  }, [response])
+
+  React.useEffect(() => {
+    setResponse(passedResponse)
+  }, [passedResponse])
+
+  function extractConceptResultsFromResponse(response) {
+    const { concept_results, } = response
+
+    if (!concept_results) { return {} }
+
+    if (typeof concept_results === 'string') {
+      return JSON.parse(concept_results)
+    } else {
+      return concept_results
+    }
+  }
```

Comment:
It would be best to move this outside of the component itself, because otherwise it's ambiguous whether `response` refers to the argument or the state variable. You could also just rename the argument, but since the function isn't using anything inside of the component itself there's no reason to have it inside anyway.